version: "3.6"

networks:
  asus:
    driver: bridge

volumes:
    mysql_data: {}
    grafana_data: {}

services:

  mysql:
    image: mariadb:latest
    container_name: mysql
    env_file:
      - .env
    volumes:
      - mysql_data:/var/lib/mysql
      - ./mysql/db:/docker-entrypoint-initdb.d
      - ./mysql/conf/eventSceduler.cnf:/etc/mysql/conf.d/eventSceduler.cnf
    environment:
      - MYSQL_ROOT_PASSWORD=${ROOT_PASS}
      - MYSQL_DATABASE=${dbName}
      - TZ=${TZ}
    ports:
      - 3306:3306
    restart: always
    healthcheck:
      test: ["CMD", "mysqladmin", "-u${monitorUser}", "-p${monitorPass}",  "ping", "-h", "localhost"]
      interval: 2s
      timeout: 20s
      retries: 10
    networks:
      - asus

  asus:
    build: ./python/
    env_file:
      - .env
    container_name: asusMonitor
    depends_on: 
      - mysql
    links:
      - mysql
    restart: on-failure
    networks:
      - asus
  
  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    env_file:
      - .env
    volumes:
      - grafana_data:/var/lib/grafana
      - ./grafana/provisioning:/etc/grafana/provisioning
    environment:
      - GF_SECURITY_ADMIN_USER=${ADMIN_USER}
      - GF_SECURITY_ADMIN_PASSWORD=${ADMIN_PASSWORD}
      - GF_USERS_ALLOW_SIGN_UP=false
      - GF_INSTALL_PLUGINS=flant-statusmap-panel,ae3e-plotly-panel
    restart: unless-stopped
    ports:
      - 3000:3000
    networks:
      - asus
    labels:
      org.label-schema.group: "monitoring"